Set/Models/GameModel.swift

```diff
@@ -122,15 +126,37 @@ struct GameModel {
         }
     }
     
-    mutating private func getBonus(for player: Player) {
+    private mutating func isFasterThanPreviousMatch() -> Bool {
+        let matchDate = Date.now
+        let matchTimeInterval = matchDate.timeIntervalSince(previousMatchDate ?? startGameDate)
+        
+        guard let previousMatchTimeInterval = previousMatchTimeInterval else {
+            self.previousMatchTimeInterval = matchTimeInterval
+            previousMatchDate = matchDate
+            return false
+        }
+        
+        self.previousMatchTimeInterval = matchTimeInterval
+        previousMatchDate = matchDate
+        
+        return matchTimeInterval < previousMatchTimeInterval
+    }
+    
+    private mutating func getBonus(for player: Player) {
+        var bonus = Constants.defaultBonus
+        
+        if isFasterThanPreviousMatch() {
+            bonus += 1
```

Comment:
Please don't leave constants like this, you already have special place where to store it, be clean from beginning.